### DISCUSSION TITLE
Current best way to build dataset or arrays with varying coords

### DISCUSSION THREAD

**Discussion by EtienneCmb** [pydata/xarray#4847] [Q&A]
Hi,

I'm working with relatively large files, all loaded as `DataArray`. All of them have the same dimension names, however, the coordinates might be different. I want to select sub-data across all the files e.g using `.sel`. My first idea was to use a `Dataset` composed of all of my files. Here's a minimal example :

```python
da_1 = xr.DataArray([1, 2, 3], dims='x', coords=([0, 1, 2],))
da_2 = xr.DataArray([3, 4, 5], dims='x', coords=([2, 3, 4],))

xr.Dataset({'x1': da_1, 'x2': da_2})
```

Missing values are filled with `nan`, I suppose that it's also going to increase quite drastically the size in memory when considering much larger files?

![image](https://user-images.githubusercontent.com/15892073/106293408-446e9100-624e-11eb-99e0-2b359da486f6.png)

In perfect world, I would like to have a container such as the `Dataset`, which doesn't create internal copies (or extend it with nan) and still allow me to sub-select data across all arrays. What would be the best strategy using xarray?

Thank you in advance,

**Comment by keewis** [comment]
I'm not quite sure but my guess would be to use `sparse` or to use different dimension names (see also #3984). Selecting would be a bit more difficult for the latter, though: you would have to translate `x` to `x1` and `x2` (assuming the new dimension names are called that), with identical indexers.
